src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,34 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws ImpossibleRegisterUserException {
+        if (user == null) {
+            throw new ImpossibleRegisterUserException("The user can`t be empty");
+        } else if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new ImpossibleRegisterUserException("This login is incorrect");
+        } else if (user.getPassword() == null || user.getPassword().length() < 6) {
```

Comment:
The password validation checks if the password is null or has a length less than 6. Verify that this matches the task requirements. If not, update the validation logic to align with the specified requirements.